Source/WebCore/platform/graphics/avfoundation/InbandMetadataTextTrackPrivateAVF.cpp

```diff
@@ -56,27 +56,35 @@ InbandMetadataTextTrackPrivateAVF::~InbandMetadataTextTrackPrivateAVF() = defaul
 
 void InbandMetadataTextTrackPrivateAVF::addDataCue(const MediaTime& start, const MediaTime& end, Ref<SerializedPlatformDataCue>&& cueData, const String& type)
 {
+    ASSERT(isMainThread());
     ASSERT(cueFormat() == CueFormat::Data);
     ASSERT(start >= MediaTime::zeroTime());
 
-    if (!client())
+    if (!hasClients())
         return;
+    ASSERT(hasOneClient());
```

Comment:
Could a client be added on another thread in between the `hasClients()` check and this assertion?